pkg/controller/controller_utils.go

```diff
@@ -223,8 +223,8 @@ type PodControlInterface interface {
 	CreatePods(namespace string, template *api.PodTemplateSpec, object runtime.Object) error
 	// CreatePodsOnNode creates a new pod accorting to the spec on the specified node.
 	CreatePodsOnNode(nodeName, namespace string, template *api.PodTemplateSpec, object runtime.Object) error
-	// DeletePod deletes the pod identified by podID.
-	DeletePod(namespace string, podID string) error
+	// DeletePod deletes the pod controlled by controller
+	DeletePod(namespace string, pod *api.Pod, object runtime.Object) error
```

Comment:
Why did you change from podID (pod.Name) to full pod?. You're using only pod.Name in the implementation.